### ISSUE TITLE
Favorites

### ISSUE THREAD

**Issue by blm07** [Herdinger/EmulationStation#20]
Would there be any way to add a favorites option to the rom selection? Like you select your favorite games, and then at the bottom of the list there is an option to go into the full rom selection.

**Comment by zigurana** [comment]
Yes, the referenced PR has favorites in place, but also a bit more than that. Actually, I took @Kaptainkia 's PR for favorites as a starting point for my work. See: https://github.com/Aloshi/EmulationStation/pull/393

**Comment by GlassedSilver** [comment]
Why stop there? Having tags or something similar would be very handy.
Folders, too.

Especially when you want to browse and discover games or narrow them down to a selection of some criteria.

You may say, well a searching algorithm would help, yes and no.
Discoverability to search is like a web directory to Google.
Yes, Google's search won over Yahoo's directory back in the day, but only because the need to quickly find a specific thing is greater than discovering and stumbling upon things you don't think about right away.
This doesn't mean the directory is a worse system, it's just different and both should compliment each other.
Now, Yahoo Directories may be dead, but are you saving webpages in your bookmarks in one big list or are they nested in folders to group them?
I don't know about you, but if I stored them alphabetically I wouldn't ~~find~~ discover jack. :)

**Comment by zigurana** [comment]
@GlassedSilver. I will not claim I fully understand your comment vis-a-vis searching vs directories, but let me say this at least: How are you going to populate the directories or tags or any other method of classification you have in mind?
Not only will my (perfectly logical) method be different from yours, but also, before you can implement the tags, you will need to 'discover' the games first! So what would this help you?

Or are you offering to implement some kind of fuzzy-logic searching based on wiki-crawling so this whole thing can be automated?

In any case, I think there is some merit in (manual) tagging, but we'll need a proper (SQL) database infrastructure before we can make that happen. Which is on the roadmap for this fork, so fingers crossed!

**Comment by GlassedSilver** [comment]
Maybe I should rephrase what I mean by discover:

Once you've amassed a nice collection of ROMs you tend to forget about some titles or you may not think about every detail every game has that you tag them with.

So in a way I mean re-discover.

And yes, I'm talking about manual organization, much like how I already invest a lot of work into keeping my iTunes collection tidy and neat and sorted and play-listed I'd love to do the same with my ROMs.
A filesystem is great, an application to show me everything at a glance is great as well, but it's really frustrating not to be able to sort things in a way that is logical for you.

It takes a minute maybe to add the most important details about a movie I rip in iTunes, but later being able to rediscover everything that's relevant to a query for a director, a specific comment or tag, ... is invaluable.

I'd love to have this feature in EmulationStation because it could propagate to whatever great products are based on the work done with this project, hopefully your fork becomes the main one eventually. (any news on that btw? I think I've read you'd not be opposed to being the canonical go-to project)

As for how I'd like to sort, well...
I have a few tags/categories I'd set up. I think of categories as mutually exclusive and tags as something that can apply in any manner. (just so we're on the same page, but I believe that's the general understanding too)

Categories: Homebrew, Released game, cancelled game/protoype, Demo/Alpha (released/official) and such
Tags: Fan-game, NetPlay (I assume that could be an automatic tag?), Spin-off, Relaxing, Anime, cute, etc... basically anything possible I may want to quickly filter for.

And great to hear that a database structure is planned anyways! It's a relief to hear that those last little bits that keep me from fully investing into ES-based setups have a good chance of getting a fix. 👍
